core/src/PlatformProxyFactory.cs

```diff
@@ -1,66 +1,80 @@
+// 
 // Copyright (c) Microsoft Corporation.
 // All rights reserved.
-//
+// 
 // This code is licensed under the MIT License.
-//
+// 
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files(the "Software"), to deal
 // in the Software without restriction, including without limitation the rights
 // to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
 // copies of the Software, and to permit persons to whom the Software is
 // furnished to do so, subject to the following conditions :
-//
+// 
 // The above copyright notice and this permission notice shall be included in
 // all copies or substantial portions of the Software.
-//
+// 
 // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
 // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-//
-//------------------------------------------------------------------------------
+// 
+// ------------------------------------------------------------------------------
 
 using System;
 
 namespace Microsoft.Identity.Core
 {
     /// <summary>
-    /// Returns the platform / os specific implementation of a PlatformProxy. 
+    ///     Returns the platform / os specific implementation of a PlatformProxy.
     /// </summary>
     internal class PlatformProxyFactory
     {
-        private PlatformProxyFactory() { }
-
         // thread safety ensured by implicit LazyThreadSafetyMode.ExecutionAndPublication
-        private static readonly Lazy<IPlatformProxy> _platformProxyLazy = new Lazy<IPlatformProxy>(() =>
+        private static readonly Lazy<IPlatformProxy> _platformProxyLazy = new Lazy<IPlatformProxy>(
+            () =>
 #if NET_CORE
-            new NetCorePlatformProxy()
+            new NetCorePlatformProxy(IsMsal())
 #elif ANDROID
-            new AndroidPlatformProxy()
+            new AndroidPlatformProxy(IsMsal())
 #elif iOS
-            new iOSPlatformProxy()
+            new iOSPlatformProxy(IsMsal())
 #elif WINDOWS_APP
-            new UapPlatformProxy()
+            new UapPlatformProxy(IsMsal())
 #elif FACADE
-            new NetStandard11PlatformProxy()
+            new NetStandard11PlatformProxy(IsMsal())
 #elif NETSTANDARD1_3
-            new Netstandard13PlatformProxy()
+            new Netstandard13PlatformProxy(IsMsal())
 #elif DESKTOP
-            new NetDesktopPlatformProxy()
+            new NetDesktopPlatformProxy(IsMsal())
 #endif
         );
 
+        private PlatformProxyFactory()
+        {
+        }
+
+        private static bool IsMsal()
```

Comment:
I think PlatformProxy does too much now. My initial refactoring idea was to break up the old "PlatformInformation" into a PlatformProxy that focuses only on OS and .net flavour (read the upn from the OS etc.) and a LibraryProxy that would handle things like "get version".

Reply:
I'd like to proceed with the current refactor since all of the code being moved in here is paltform specific.  Personally, I like the idea of having some root object (like platform proxy) that gives is the central core for how we do platform disambiguation.  But if we want to break this down into smaller parts hanging off platformproxy or something else, let's look into that.